elyra/cli/pipeline_app.py

```diff
@@ -300,5 +301,66 @@ def run(pipeline_path):
     print_banner("Elyra Pipeline Local Run Complete")
 
 
+@click.command()
+@click.option('--json',
+              'json_option',
+              is_flag=True,
+              required=False,
+              help='Returns pipeline file in a machine-readable JSON format')
+@click.argument('pipeline_path')
+def describe(json_option, pipeline_path):
+    """
+    Display pipeline description in a human readable format
+    """
+
+    _validate_pipeline_file_extension(pipeline_path)
+
+    pipeline_definition = \
+        _preprocess_pipeline(pipeline_path, runtime='local', runtime_config='local')
```

Comment:
Forcing a `local` runtime (and runtime_config) causes the preprocessing to fail on non-generic pipelines (i.e., pipelines that are specific to either `airflow` or `kfp`) because `local` doesn't match the runtime validation within the preprocessing and leads to this:
```
$ elyra-pipeline describe ~/notebooks/elyra/my_pipeline/my_kfp_only.pipeline 
Traceback (most recent call last):
  File "/opt/anaconda3/envs/elyra-dev/bin/elyra-pipeline", line 8, in <module>
    sys.exit(pipeline())
  File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/elyra/cli/pipeline_app.py", line 319, in describe
    _preprocess_pipeline(pipeline_path, runtime='local', runtime_config='local')
  File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/elyra/cli/pipeline_app.py", line 140, in _preprocess_pipeline
    runtime_description = primary_pipeline['app_data']['ui_data']['runtime']['display_name']
KeyError: 'runtime'
```
I would change those `runtime` and `runtime_config` parameters to `None` for now and this issue will need to be resolved in another PR.

There should be a way to discover the runtime name or separate that validation out of the preprocessing so that options like`describe` can just display what's in the pipeline.

I'm opening a separate issue for the validation.

cc: @lresende

Reply:
@19690ao - I've addressed the above by allowing the `runtime` and `runtime-config` parameters to be optional.  Please rebase your changes against the master branch and update your call to `_preprocess_pipeline()` to only take the pipeline filename.  You should find the runtime-config value via `pipeline_data['app_data'].get('runtime')`.  Note that this value may be None if the runtime is generic.  The actual runtime name should be located via `pipeline_data['app_data']['properties'].get('runtime')`.  This value should exist for all pipelines and will be one of 'Generic', 'Kubeflow Pipelines', and 'Apache Airflow'.

Reply:
@kevin-bates - Thanks. I've rebased my local branch. I'm still having an issue describing non-generic pipelines.

> ... update your call to `_preprocess_pipeline()` to only take the pipeline filename.

Currently, if I change that line to only take the `pipeline_path` as an argument, running a kfp file through the describe command is currently giving me this output:
```
Error: This pipeline requires an instance of Kubeflow Pipelines runtime configuration.
The specified configuration 'None' is for None runtime.  
```
I was under the impression that the changes made were meant to allow the _preprocess_pipeline function to work without the need of specifying the runtime/runtime_config. I'm not quite sure how to run the function at the moment.

Reply:
Ok - I'll need to poke around at this relative to your PR.  It looks like `_validate_pipeline_runtime()` needs to flip its preference and default to returning True and be strict with its definition of _invalid_.  I will come up with a definition of the function that you can use and include in your PR.  It's critical that `describe` not have to know about the runtime.